bazel/cc_wrapper.py

```diff
@@ -42,18 +43,17 @@ def main():
   if ("-static-libstdc++" in sys.argv[1:] or "-stdlib=libc++" in sys.argv[1:] or
       "-std=c++0x" in sys.argv[1:]):
```

Comment:
Previously, this code path was used only for detecting cases with non-interchangeable `CC` and `CXX` for statically linked libstdc++, and later libc++. Those checks are not sufficient for detecting C vs C++ targets, and I imagine that, when used as-is, this code would leak some `CFLAGS` into C++ builds, and possibly break them.

At the very least, I think that you need to add checks for other `-std=` C++ values (starting with `-std=c++`  and `-std=gnu++` might be enough), `-lstdc++`, `-lc++`, and update the comment above this code block.